src/components/_helper-functions/helpers.tsx

```diff
@@ -33,40 +20,38 @@ export const generateInitialFormValues = ({
     serialNumber: ""
   };
 
-  // const initialFormValues: EndUserDeploymentFormField = {
-  //   endUser: "",
-  //   ticketNumber: "",
-  //   items: []
-  // };
+  const initialFormValues: EndUserDeploymentFormField = {
+    endUser: "",
+    ticketNumber: "",
+    items: []
+  };
 
   optionsState.formCount.forEach(() => {
-    const values = new InitialFormValues();
+    // const values = new InitialFormValues();
     // const values = initialFormValues;
-    vals.push(values);
+    const initialFormValue = { ...initialFormValues, items: [] }
```

Comment:
This was the main cause. When using spread on an object with an array in it it's going to keep the reference of the original array. So in addition to spreading here I'm also supplying it with a new array for items.